diff --git a/node_modules/@astrojs/starlight/components/ThemeSelect.astro b/node_modules/@astrojs/starlight/components/ThemeSelect.astro
index a7cd175..d3e9670 100644
--- a/node_modules/@astrojs/starlight/components/ThemeSelect.astro
+++ b/node_modules/@astrojs/starlight/components/ThemeSelect.astro
@@ -48,9 +48,20 @@ import Select from './Select.astro';
 
 	/** Update select menu UI, document theme, and local storage state. */
 	function onThemeChange(theme: Theme): void {
-		StarlightThemeProvider.updatePickers(theme);
-		document.documentElement.dataset.theme = theme === 'auto' ? getPreferredColorScheme() : theme;
-		storeTheme(theme);
+	  StarlightThemeProvider.updatePickers(theme);
+	
+	  const resolvedTheme = theme === 'auto' ? getPreferredColorScheme() : theme;
+	  const event = new CustomEvent('starlight:theme-change', {
+	    cancelable: true,
+	    detail: { theme, resolvedTheme },
+	  });
+  
+	  const handled = !document.dispatchEvent(event);
+	  if (!handled) {
+	    document.documentElement.dataset.theme = resolvedTheme;
+	  }
+  
+	  storeTheme(theme);
 	}
 
 	// React to changes in system color scheme.
